packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -519,7 +562,53 @@ class RenderWrap extends RenderBox
     int childCount = 0;
     RenderBox? child = firstChild;
     while (child != null) {
-      final Size childSize = layoutChild(child, childConstraints);
+      Size childSize;
+
+      final WrapParentData childParentData = child.parentData! as WrapParentData;
+      WrapFit fit = childParentData.fit;
+
+      if (mainAxisLimit.isInfinite) {
+        switch (fit) {
+          case WrapFit.runTight:
+            fit = WrapFit.runLoose;
+          case WrapFit.tight:
+            fit = WrapFit.loose;
+          case WrapFit.loose:
+          case WrapFit.runLoose:
+        }
+      }
+
+      switch (fit) {
+        case WrapFit.runTight:
+        case WrapFit.runLoose:
+          final double childIntrinsicMainAxisExtent = getChildMinIntrinsicMainAxisExtent(child, double.infinity);
+          if (runMainAxisExtent != 0 && runMainAxisExtent + childIntrinsicMainAxisExtent + spacing <= mainAxisLimit) {
```

Comment:
We probably shouldn't invoke (and pay the cost of) intrinsics if runMainAxisExtent is zero since it remains unused?

Reply:
Good catch. eb28ae48db should have fixed this.